main.go

```diff
@@ -133,6 +133,17 @@ func main() {
 	}
 
 	setupFinished := make(chan struct{})
+	go func() {
```

Comment:
can you explain in the commit message why we had to move this? It's really not obvious, as the goroutine should wait the channel, so the order should not matter.
And, iirc we "took inspiration" from the metallb's code, would it make sense to apply the change there too?

Reply:
I think what happens in CI is getting fixed here by incident (probably because CI is running on a single core?).
What we really need to do is to ensure that the operator is ready.
In metallb, together with changing the order of the goroutines, we also had to wait on the ready channel before listening on the readiness probe. We should do the same here, to make it bullet proof.

Reply:
sure, done.